app/screens/invite/selection.tsx

```diff
@@ -0,0 +1,343 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import React, {useCallback, useState, useMemo} from 'react';
+import {
+    Keyboard,
+    Platform,
+    View,
+    LayoutChangeEvent,
+    useWindowDimensions,
+    FlatList,
+    ListRenderItemInfo,
+    ScrollView,
+} from 'react-native';
+import Animated, {useDerivedValue} from 'react-native-reanimated';
+import {useSafeAreaInsets} from 'react-native-safe-area-context';
+
+import SelectedUser from '@components/selected_users/selected_user';
+import TouchableWithFeedback from '@components/touchable_with_feedback';
+import UserItem from '@components/user_item';
+import {MAX_LIST_HEIGHT, MAX_LIST_TABLET_DIFF} from '@constants/autocomplete';
+import {useTheme} from '@context/theme';
+import {useAutocompleteDefaultAnimatedValues} from '@hooks/autocomplete';
+import {useIsTablet, useKeyboardHeight} from '@hooks/device';
+import {makeStyleSheetFromTheme, changeOpacity} from '@utils/theme';
+
+import {SearchResult} from './invite';
+import SelectedEmail from './selected_email';
+import SelectionSearchBar from './selection_search_bar';
+import SelectionTeamBar from './selection_team_bar';
+import TextItem, {TextItemType} from './text_item';
+
+const AUTOCOMPLETE_ADJUST = 5;
+const KEYBOARD_HEIGHT_ADJUST = 3;
+
+const INITIAL_BATCH_TO_RENDER = 15;
+const SCROLL_EVENT_THROTTLE = 60;
+
+const keyboardDismissProp = Platform.select({
+    android: {
+        onScrollBeginDrag: Keyboard.dismiss,
+    },
+    ios: {
+        keyboardDismissMode: 'on-drag' as const,
+    },
+});
+
+const keyExtractor = (item: SearchResult) => (
+    typeof item === 'string' ? item : (item as UserProfile).id
+);
+
+const getStyleSheet = makeStyleSheetFromTheme((theme: Theme) => {
+    return {
+        container: {
+            display: 'flex',
+            flex: 1,
+        },
+        searchList: {
+            left: 20,
+            right: 20,
+            position: 'absolute',
+            bottom: Platform.select({ios: 'auto', default: undefined}),
+        },
+        searchListBorder: {
+            borderWidth: 1,
+            borderColor: changeOpacity(theme.centerChannelColor, 0.2),
+            borderRadius: 4,
+            elevation: 3,
+        },
+        searchListPadding: {
+            paddingVertical: 8,
+        },
+        searchListShadow: {
+            shadowColor: '#000',
```

Comment:
To help with the layout calculation setting a background color always help. If the background color mess things up, leave it out

Reply:
Good tip, added.

Reply:
I don't see it

Reply:
Added in `searchList`.

Reply:
it needs to be on the same style, if it is in a different style it does not get applied in the calculation. So you need to move it to this style and test it, if it does not work correctly, feel free to remove it.

Reply:
Done.